Add VPCPeering to subctl

ruromero opened the pull request:
Signed-off-by: ruromero <rromerom@redhat.com>

Related to: 
* https://github.com/submariner-io/enhancements/issues/64

Depends on https://github.com/submariner-io/cloud-prepare/pull/190

Adding the `subctl cloud vpc-peering <cloud> [flags]` command for setting a VPC Peering between two clusters.

The vpc-peering command must be executed after the `cloud prepare` command to make sure all the pre-requisites are met.

As it requires a _target_ cloud to also be able to connect to it to perform all the necessary actions, the flag names will be the same but prefixed with _target_ i.e. `--target-region`. The same flags have been added so that different profiles or credentials file can be used.

Some usage examples:

* With metadata-file
```
$ subctl cloud vpc-peering aws --metadata-file=source-aws-cloud.json --target-metadata-file=target-aws-cloud.json
```
* With all other flags
```
$ subctl cloud vpc-peering aws --region=eu-west-1 --infra-id=source-infra --target=region=eu-west-2 --target-infra-id=target-infra
```

skitt reviewed (changes requested):
Thanks for taking care of this!

Could you add some more explanations to the commit message, especially around the flag handling and the changes that will introduce for subctl users?

ruromero commented:
@skitt done

dfarrell07 commented:
As I think @skitt said on Slack, the details of the change should be in the commit message, not just the PR description.

https://github.com/submariner-io/submariner-operator/pull/1794/commits/be183e7a2045c9fd223c70dfcd92f693b1e257b5 (still showing only basic info here)

ruromero commented:
@dfarrell07 updated

dfarrell07 commented:
Thanks @ruromero! There are a number of test jobs failing that need attention. Some of them should provide info inline, here are some additional details for the commit message linter if you're interested: https://submariner.io/development/code-review/#commit-message-formatting

ruromero commented:
As this PR depends on another one which is still WIP I won't worry about fixing them ATM. I just wanted to get some early feedback about the solution provided, the code style, the naming, etc.

dfarrell07 commented:
We have some automation to help track dependencies like that. :nerd_face: 

https://github.com/submariner-io/submariner-operator/blob/devel/.github/workflows/dependent-issues.yml

Depends on https://github.com/submariner-io/cloud-prepare/pull/190

Update: but it doesn't trigger for PR comments I see (which makes sense, that would be a lot of jobs), it should on "edited" events though.

Jaanki commented on pkg/subctl/cmd/cloud/vpcpeering/aws.go:
```diff
@@ -0,0 +1,68 @@
+	if err != nil {
+		reporter.Failed(err)
+
+		utils.ExitOnError("Failed to initialize AWS connectivity", err)
```
```suggestion
		exit.OnErrorWithMessage("Failed to initialize AWS connectivity", err)
```

Also applies to other places.

ruromero commented:
we're still working on this

ruromero committed:
Add vpc-peering to subctl

Use subctl cloud vpc-peering create <cloud> [flags] command for setting a VPC
Peering between two clusters.

The vpc-peering command must be executed after the cloud prepare command to
make sure all the pre-requisites are met.

As it requires a target cloud to also be able to connect to it to perform all
the necessary actions, the flag names will be the same but prefixed with
target i.e. --target-region. The same flags have been added so that different
profiles or credentials file can be used.

Some usage examples:

* With metadata-file
$ subctl cloud vpc-peering create aws --metadata-file=source-aws-cloud.json \
 --target-metadata-file=target-aws-cloud.json

* With all other flags
$ subctl cloud vpc-peering create aws --region=eu-west-1 --infra-id=source-infra \
 --target=region=eu-west-2 --target-infra-id=target-infra

The cleanup method requires the same arguments:

e.g.
$ subctl cloud vpc-peering cleanup aws --metadata-file=source-aws-cloud.json \ --target-metadata-file=target-aws-cloud.json

Signed-off-by: ruromero <rromerom@redhat.com>

Signed-off-by: Alejandro Villegas <alex.ansi.c@gmail.com>

ruromero commented:
@skitt it should be ready for review. Once the changes and tests look good I can squash all the commits. Let me know if you prefer me to do it before.
Thanks

skitt commented on .gitignore:
```diff
@@ -16,4 +16,4 @@ vendor/
 Makefile.dapper
 Dockerfile.*
-
+.*.swp
```
Please [configure your local gitignore](https://sebastiandedeyne.com/setting-up-a-global-gitignore-file/) instead of adding environment-specific entries to the project’s `.gitignore`.

skitt commented:
> @skitt it should be ready for review. Once the changes and tests look good I can squash all the commits. Let me know if you prefer me to do it before. Thanks

Thanks, unless the commits are significant for review, I’d rather have them squashed sooner than later. We’re in the middle of the 0.12 release now, I’ll give this the attention it deserves next week!

ruromero commented:
@skitt I have rebased and squashed all commits

dfarrell07 commented:
Are we still planning on working on this?

dfarrell07 commented:
This will need to move to the new subctl repo.
